src/components/profile.ts

```diff
@@ -0,0 +1,164 @@
+const yearStart = 1900;
+const yearEnd = new Date().getFullYear() + 7;
+
+export enum configMaps {
+  aboutme = 'about_me',
+  birthdate = 'birth_date',
+  preferredname = 'preferred_name',
+  preferredpronouns = 'preferred_pronouns',
+  term = 'term',
+  year = 'year',
+  faculty = 'faculty',
+  program = 'program',
+  specialization = 'specialization'
+}
+
+const validBirthdates = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December'
+];
+
+export const customizationLimits = {
+  aboutme:
+    'Feel free to enter anything you want about yourself! Your hobbies, interests, etc. Must be less than 1000 characters.',
+  birthdate:
+    'To protect your personal information, this must only be your month of birth. Must be one of ' +
+    validBirthdates.join(', '),
+  preferred_name: 'Your preferred name.',
+  preferred_pronouns: 'Your preferred pronouns.',
+  term: 'Your school term. Must be one of ' + validTerms.join(', '),
+  year: `The year you will graduate. Must be between ${yearStart} and ${yearEnd}`,
+  faculty: 'Your faculty. Must be one of ' + validFaculties.join(', '),
+  program: 'Your program.',
+  specialization: 'Your specialization.'
+};
+
+interface userCustomization {
+  reason: string;
+  parsedDescription?: string;
+}
+
+enum validatedFields {
+  birthdate = 'birthdate',
+  term = 'term',
+  faculty = 'faculty',
+  year = 'year'
+}
+
+export const validCustomizations = `${(Object.keys(configMaps) as Array<
+  keyof typeof configMaps
+>).map((key) => ' ' + key)}`
+
+export const validUserCustomization = (
+  customization: keyof typeof configMaps,
+  description: string
+): userCustomization => {
+  if (!(Object.keys(configMaps) as Array<keyof typeof configMaps>).includes(customization)) {
+    return {
+      reason: `Invalid customization selection. Must be one of**${validCustomizations}**`
+    };
+  }
+  let parsedDescription = description;
+  if (customization !== validatedFields.term) {
+    // convert first letter to capital in case the user doesnt use proper capitalization
+    parsedDescription = description.toLowerCase();
+    parsedDescription = parsedDescription[0].toUpperCase() + parsedDescription.slice(1);
+  }
+  switch (customization) {
+    case validatedFields.birthdate:
+      if (!validBirthdates.includes(parsedDescription)) {
+        return { reason: 'Invalid birthdate. Must be one of : ' + validBirthdates.join(', ') + '.' };
+      }
+      break;
+    case validatedFields.term:
+      // MASc is the only term that is not full upper case
+      let checkDescription = description.toUpperCase()
+      if (checkDescription.toLowerCase() === 'MASC') {
+        parsedDescription = 'MASc';
+      } else if (checkDescription.toLowerCase() === 'PHD') {
+        parsedDescription = 'PhD';
+      } else {
+        parsedDescription = description.toUpperCase();
+      }
```

Comment:
```suggestion
      let parsedDescription = description.toUpperCase();
      if (parsedDescription === 'MASC') {
        parsedDescription = 'MASc';
      } else if (parsedDescription === 'PHD') {
        parsedDescription = 'PhD';
      }
```
I don't think the old code works?